homeassistant/components/waze_travel_time/sensor.py

```diff
@@ -197,41 +203,37 @@ async def async_added_to_hass(self) -> None:
             await self.first_update()
 
     @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
-
-    @property
-    def native_value(self):
+    def native_value(self) -> float | None:
         """Return the state of the sensor."""
         if self._waze_data.duration is not None:
             return round(self._waze_data.duration)
 
         return None
 
     @property
-    def native_unit_of_measurement(self):
-        """Return the unit of measurement."""
-        return TIME_MINUTES
-
-    @property
-    def icon(self):
+    def icon(self) -> str:
         """Icon to use in the frontend, if any."""
-        return ICON
+        if self._waze_data.vehicle_type == "taxi":
+            return "mdi:taxi"
+        if self._waze_data.vehicle_type == "motorcycle":
+            return "mdi:motorbike"
+
+        return "mdi:car"
```

Comment:
This is a change in the behavior (and might also be considered a breaking change), 
I think this should fit under code quality improvement and implemented on a separate PR

Reply:
I agree